### PR TITLE
Add support for SolrInputDocument and related save operations #3

### PR DESCRIPTION
**Pull request by cponomaryov**
Adds support for all supported by SolrJ operations in both APIs (Scala and Java):
- add docs
- add doc
- add bean
- add beans
- commit
- optimize
- rollback
- delete by id
- delete by ids
- delete by query
- ping
- query
- query and stream response
- get by id
- get by ids

### CODE REVIEW

**Comment by magro** [comment]
That's awesome, many thanks! I had a quick look at it (on my phone only) and so far everything looked good. 
Do you think we should update the README somewhere?

**Comment by cponomaryov** [comment]
Thanks! I'll take care of documentation tomorrow.

**Comment by cponomaryov** [comment]
I updated the README, take a look please.

File: README.md [hunk 1]
```diff
@@ -4,14 +4,16 @@
...
+This is a java/scala solr client providing an interface like SolrJ, just asynchronously / non-blocking
 (built on top of [async-http-client](https://github.com/AsyncHttpClient/async-http-client) / [netty](https://github.com/netty/netty)).
 For java it supports `CompletableFuture`, for scala you can choose between twitter's `Future` or the standard/SDK `Future`.
 
 ## Contents
 
 - [Installation](#installation)
 - [Usage](#usage)
+ - [Adding Data to Solr](#adding-data-to-solr)
+   - [Directly adding POJOs to Solr](#directly-adding-pojos-to-solr)
```
Comment (magro) [inline README.md]: We probably don't need this level/item in the TOC, WDYT?

File: README.md [hunk 2]
```diff
@@ -127,6 +129,112 @@ val solr = AsyncSolrClient.Builder("http://localhost:8983/solr/collection1")
...
+    .thenAccept(r -> System.out.println("beans added"));
+```
+
+[scala]
+```scala
+import io.ino.solrs.AsyncSolrClient
+import io.ino.solrs.future.ScalaFutureFactory.Implicit // or TwitterFutureFactory
+import org.apache.solr.client.solrj.beans.Field;
+
+case class Item(@(Field@field) id: String,
```
Comment (magro) [inline README.md]: Needs an `import scala.annotation.meta.field`? Maybe add a space between `Field` and `@field`?

**Comment by cponomaryov** [comment]
Done

**Comment by magro** [comment]
Awesome, thanks!

**Comment by magro** [comment]
I just released this as 2.0.0-RC4, it's on its way to maven central.

**Comment by cponomaryov** [comment]
Thanks!

**Pull request merged**